lib/ansible/playbook/conditional.py

```diff
@@ -181,8 +181,15 @@ def generic_visit(self, node, inside_call=False, inside_yield=False):
                 raise AnsibleError("Invalid conditional detected: %s" % to_native(e))
 
             # and finally we generate and template the presented string and look at the resulting string
+            # NOTE The spaces around True and False are intentional to short-circuit safe_eval and avoid
+            #      its expensive calls.
             presented = "{%% if %s %%} True {%% else %%} False {%% endif %%}" % conditional
-            val = templar.template(presented, disable_lookups=disable_lookups).strip()
+            # NOTE Convert the result to text to account for both native and non-native jinja.
+            # NOTE The templated result of `presented` is string on native jinja as well prior to Python 3.10.
+            #      ast.literal_eval on Python 3.10 removes leading whitespaces so " True " becomes bool True
+            #      as opposed to Python 3.9 and lower where the same would result in IndentationError and
+            #      string " True " would be returned by Templar.
+            val = to_text(templar.template(presented, disable_lookups=disable_lookups)).strip()
```

Comment:
Can you add a test to cover this case?

Reply:
Unit test added.

Reply:
I am not sure why it fails. It works on my machine! (tm) :-)

These two passes (3.8, 3.9, 3.10):
`ansible-test units --docker --python 3.10 test/units/template/test_native_concat.py`
`ansible-test units --docker --python 3.10 test/units/template/`

This, as run in CI, fails (3.8, 3.9, 3.10):
`ansible-test units --docker --python 3.10 --changed`

Any suggestions?

Reply:
I'm guessing that `ansible.template` has been previously imported with `DEFAULT_JINJA2_NATIVE` not enabled.

I've had to jump through hoops before with forcing re-evaluation of imports in units.  I don't know if others think this is acceptable:

```
def test_cond_eval(monkeypatch):
    monkeypatch.delitem(sys.modules, 'ansible.template')
    monkeypatch.setattr(C, 'DEFAULT_JINJA2_NATIVE', True)
    mod = importlib.import_module('ansible.template')
    Templar = mod.Templar
    AnsibleNativeEnvironment = mod.AnsibleNativeEnvironment

    # Test code here
```

Reply:
Since the unit tests depend on setting `DEFAULT_JINJA2_NATIVE` before certain imports, the test will need to use a fixture to remove existing imports (if any) from `sys.modules`, set `DEFAULT_JINJA2_NATIVE` and then perform the imports.

This is needed since other tests may have already performed those imports. To see the failure from CI, try running all the unit tests -- or at least include one that uses the same imports and runs before the failing test.

Reply:
BTW, I did test that, and it seems to pass all units.

Reply:
> I'm guessing that ansible.template has been previously imported with DEFAULT_JINJA2_NATIVE not enabled.

That was my guess as well but the fact that `ansible-test units --docker --python 3.10 test/units/template/` worked confused me, but it seems like `test_native_concat.py` just runs first from that directory.

Thanks, I'll play with this.